nvfuser/__init__.py

```diff
@@ -50,6 +51,54 @@ def disable_automatic_serialization():
     atexit.unregister(_C.serialize)
 
 
+class DistributedTensor(torch.Tensor):
+    """Wraps a _C._DistributedTensor as a torch.Tensor.
+
+    This way, the user can use the underlying local tensor without extra conversion.
+    For example, `torch.testing.assert_close(dtensor, expected_local_tensor)`.
+    """
+
+    _dtensor: _C._DistributedTensor
+
+    @staticmethod
+    def __new__(cls, dtensor: _C._DistributedTensor):
+        t = dtensor.local
+        return torch.Tensor._make_wrapper_subclass(
+            cls,
+            t.shape,
+            strides=t.stride(),
+            storage_offset=t.storage_offset(),
+            device=t.device,
+            layout=t.layout,
+            requires_grad=t.requires_grad,
+            dtype=t.dtype,
+        )
+
+    def __init__(self, dtensor: _C._DistributedTensor):
+        self._dtensor = dtensor
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+        def unwrap(t):
+            if isinstance(t, DistributedTensor):
+                return t._dtensor.local
+            return t
+
+        return func(*tree_map(unwrap, args), **tree_map(unwrap, kwargs))
```

Comment:
Per https://pytorch.org/docs/stable/notes/extending.html#subclassing-torch-tensor

I think this should be 
```suggestion
        return super().__torch_function__(func, types, *tree_map(unwrap, args), **tree_map(unwrap, kwargs))
```

Reply:
I couldn't get `__torch_function__` to work, so cc @syed-ahmed to advise. AFAICT, to wrap `_C._DistributedTensor` as a `torch.Tensor`, I need `torch.Tensor._make_wrapper_subclass`. And `torch.Tensor._make_wrapper_subclass` requires `__torch_dispatch__`. However, the [doc](https://pytorch.org/docs/stable/notes/extending.html#subclassing-torch-tensor) @jjsjann123 pointed me to seem to prefer `__torch_function__` as long as it's for Python only.